packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -448,7 +448,7 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
 
   private _renderInput(): React.ReactElement<React.HTMLAttributes<HTMLInputElement>> {
     const inputProps = getNativeProps<React.HTMLAttributes<HTMLInputElement>>(this.props, inputProperties, ['defaultValue']);
-    const ariaLabelledBy = this.props['aria-labelledby'] || (this.props.label ? this._labelId : undefined);
+    const ariaLabelledBy = this.props['aria-labelledby'] || (this.props.label && !this.props.onRenderLabel ? this._labelId : undefined);
```

Comment:
Interesting issue, but I'm concerned this might break some cases too--I would *guess* that a common use case for `onRenderLabel` involves adding some wrapper elements then calling `defaultRender`, which would render an element with `this._labelId`. Unfortunately there's not a good way to find out how common it is to use `defaultRender` or not.

I wonder if a better fix would be to update the documentation for `onRenderLabel` to note that if you *don't* call `defaultRender`, you should give your custom-rendered label an ID and set the TextField's `aria-labelledby` prop to that ID. (And update the custom label example to follow that guidance.)

Reply:
That makes sense to me, I'll take a look at that approach - thanks for the feedback!